Improve handling of environment variables

paulapreuss opened the pull request:
Explicitly ask the user to set the environment variables via an `.env` file and read the variables in the settings.

paulapreuss committed:
Import environment variables from .env file

paulapreuss committed:
Update README

paulapreuss committed:
Update requirements

Commit:
remove os and ast from settings

stefansc1 commented on app/epa/settings.py:
line endings in file are Windows CR/LF, maybe convert it to Unix LF?
Also, is there a reason for timezone Europe/Copenhagen or can this be changed to Europe/Berlin?

Bachibouzouk replied:
The timezone can be changed. For the line endings is it possible for windows user to do so? As I have Unix my line endings are by default LF

stefansc1 replied:
Changed timezone and line breaks.

stefansc1 commented on app/epa/settings.py:
```diff
@@ -37,16 +48,15 @@
-SECRET_KEY = os.getenv(
-    "EPA_SECRET_KEY", "v@p9^=@lc3#1u_xtx*^xhrv0l3li1(+8ik^k@g-_bzmexb0$7n"
+SECRET_KEY = env(
+    "EPA_SECRET_KEY", default="v@p9^=@lc3#1u_xtx*^xhrv0l3li1(+8ik^k@g-_bzmexb0$7n"
```
Does it make sense to have a fixed secret key?

Bachibouzouk replied:
Probably not, I think it is legacy code from the first fork

Bachibouzouk replied:
```suggestion
    "EPA_SECRET_KEY"

```

stefansc1 replied:
default secret key is okay, but I added an entry in README to include `EPA_SECRET_KEY` in .env file

stefansc1 commented on app/epa/settings.py:
```diff
@@ -37,16 +48,15 @@
-]
+trusted_host = env("TRUSTED_HOST", default=None)
+if trusted_host is not None:
+    CSRF_TRUSTED_ORIGINS = [f"https://{trusted_host}", f"http://{trusted_host}"]
```
Suggestion: `CSRF_TRUSTED_ORIGINS = env.list('CSRF_TRUSTED_ORIGINS')` -> looks like this in env: `localhost,127.0.0.1` or similar. This avoids trusting unsecure connection.

Bachibouzouk replied:
What I like with `"TRUSTED_HOST"` is that there is only one trusted host, whereas with your suggestion I could provide a list

stefansc1 replied:
CSRF_TRUSTED_ORIGINS is _always_ a list. Previously, only one host was specified (or only one host could be specified), but internally, this was secretly converted into two trusted origins: one HTTPS and one unencrypted HTTP. With my suggestion, this hidden conversion (and, in my opinion, security vulnerability) would disappear. Instead, all trusted origins can and must now be specified explicitly.

Bachibouzouk replied:
You got a point here. I suggest you implement your suggestions in this PR (@paulapreuss was done with her part on it)

Bachibouzouk commented on app/epa/settings.py:
```diff
@@ -37,16 +48,15 @@
-SECRET_KEY = os.getenv(
-    "EPA_SECRET_KEY", "v@p9^=@lc3#1u_xtx*^xhrv0l3li1(+8ik^k@g-_bzmexb0$7n"
+SECRET_KEY = env(
+    "EPA_SECRET_KEY", default="v@p9^=@lc3#1u_xtx*^xhrv0l3li1(+8ik^k@g-_bzmexb0$7n"
```
```suggestion
    "EPA_SECRET_KEY"

```

Commit:
remove default secret key and change timezone to Berlin

Commit:
settings changes

- revert secret key: give default. Update README to include EPA_SECRET_KEY in .env
- change TRUSTED_HOST to evaulate as list (more than one trusted host possible, no more secret conversion to HTTP). Change README to reflect this.

Commit:
settings: change line break from CRLF to LF

Bachibouzouk commented on app/epa/settings.py:
```diff
@@ -1,265 +1,265 @@
-    "queue_limit": 50,
-    "orm": "default",
-}
+"""
```
Actually this change could be perfomed projectwide in a single commit in another PR. 
How to make sure then that windows user do not reintroduce CRLF in the future?

Bachibouzouk commented on README.md:
```diff
@@ -25,20 +25,21 @@ Prior to be able to develop locally, you might need to install postgres, simply
 SQL_HOST=localhost
 SQL_PORT=5432
 DEBUG=(True|False)
+MVS_HOST_API=https://mvs-open-plan.rl-institut.de
```
```suggestion
MVS_HOST_API=<the simulation server you wish to use>
```
I prefer not having this explicitly in the README

Bachibouzouk reviewed (approved):
I tested it locally and it ran, so it is ok for me

stefansc1 committed:
Update README.md

Co-authored-by: Pierre Francois <pierre-francois.duc@rl-institut.de>
